crates/rune/src/any.rs

```diff
@@ -72,11 +70,36 @@ pub use rune_macros::Any;
 ///     name: String,
 /// }
 /// ```
-pub trait Any: Named + any::Any {
+pub trait Any: Named {
     /// The type hash of the type.
     ///
     /// TODO: make const field when `TypeId::of` is const.
     fn type_hash() -> Hash;
+
+    /// The unique id of the type.
+    fn type_id() -> TypeId;
+}
+
+/// A TypeId represents a globally unique identifier for a type.
+#[derive(PartialEq, Eq)]
+pub enum TypeId {
+    /// A statically known type id, provided by a type implementing the
+    /// core::any::Any trait.
+    Static(core::any::TypeId),
+    /// A manually specified, Universally Unique Identifer (UUID).
+    Uuid(uuid::Uuid),
```

Comment:
Not sure if this is the right approach. I essentially just copied what you would do in [Bevy](https://github.com/bevyengine/bevy/blob/main/crates/bevy_asset/src/loader.rs#L53).

It would be nice to at least check for duplicate ids when installing new types into a module.